int64 period
---
int64 period